amenity-points.mss

```diff
@@ -4,1894 +4,1243 @@
-  [feature = 'amenity_bank'][zoom >= 17] {
-    marker-file: url('symbols/bank.svg');
-    marker-fill: @amenity-brown;
-    marker-placement: interior;
-    marker-clip: false;
-  }
+    // second pass for icons without text where there is still space
+    a/marker-placement: interior;
+    a/marker-clip: false;
 
-  [feature = 'amenity_bar'][zoom >= 17] {
-    marker-file: url('symbols/bar.svg');
-    marker-fill: @amenity-brown;
-    marker-placement: interior;
-    marker-clip: false;
+    [zoom >= 17][zoom < 18] {
+      shield-file: url('symbols/fountain_z17.svg');
+      a/marker-file: url('symbols/fountain_z17.svg');
+      shield-text-dy: 8;
+    }
+    [zoom >= 18] {
+      shield-file: url('symbols/fountain.svg');
+      a/marker-file: url('symbols/fountain.svg');
+      shield-text-dy: 11;
+    }
   }
+  [feature = 'amenity_parking'][way_pixels > 900],
+  [feature = 'amenity_bicycle_parking'][way_pixels > 900],
+  [feature = 'amenity_motorcycle_parking'][way_pixels > 900] {
+    [zoom < 10] {
+      marker-placement: interior;
 
-  [feature = 'amenity_bicycle_rental'][zoom >= 17] {
-    marker-file: url('symbols/rental_bicycle.svg');
-    marker-fill: @transportation-icon;
+      [feature = 'amenity_parking'] {
+        marker-file: url('symbols/parking.svg');
+      }
+      [feature = 'amenity_bicycle_parking'] {
+        marker-file: url('symbols/bicycle_parking.svg');
+      }
+      [feature = 'amenity_motorcycle_parking'] {
+        marker-file: url('symbols/motorcycle_parking.svg');
+      }
+      [access != ''][access != 'permissive'][access != 'yes'] {
+        marker-opacity: 0.33;
+      }
+    }
+    [zoom >= 10] {
+      shield-placement: interior;
+      shield-clip: false;
+      shield-name: "[name]";
+      shield-size: @standard-text-size;
+      shield-face-name: @standard-font;
+      shield-halo-radius: @standard-halo-radius;
+      shield-halo-fill: @standard-halo-fill;
+      shield-wrap-width: @standard-wrap-width;
+      shield-fill: @transportation-text;
+      shield-text-dy: 11;
+      shield-unlock-image: true;
+
+      // second pass for icons without text where there is still space
+      a/marker-placement: interior;
+      a/marker-clip: false;
+      a/marker-fill: @transportation-icon;
+
+      [access != ''][access != 'permissive'][access != 'yes'] {
+        shield-opacity: 0.33;
+        shield-text-opacity: 0.33;
+        a/marker-opacity: 0.33;
+      }
+
+      [feature = 'amenity_parking'] {
+        shield-file: url('symbols/parking.svg');
+        a/marker-file: url('symbols/parking.svg');
+      }
+      [feature = 'amenity_bicycle_parking'] {
+        shield-file: url('symbols/bicycle_parking.svg');
+        a/marker-file: url('symbols/bicycle_parking.svg');
+      }
+      [feature = 'amenity_motorcycle_parking'] {
+        shield-file: url('symbols/motorcycle_parking.svg');
+        a/marker-file: url('symbols/motorcycle_parking.svg');
+      }
+    }
+  }
+
+  // markers without text
```

Comment:
I'm not sure how a shield without text behaves and we would still have the text-dy which probably takes up space even if name is empty. I'm also not sure if a shield symbolizer uses more resources so it might be not good to use it unnecessarily.

But I'm open to other ideas.

Another option would be to solve this cartographically by not having POI icons where their label shows up at later zoom level. But I would be against that.